mock/py/mockbuild/mounts.py

```diff
@@ -85,17 +85,29 @@ def __repr__(self):
 class BindMountPoint(MountPoint):
     '''class for managing bind-mounts in the chroot'''
     @traceLog()
-    def __init__(self, srcpath, bindpath):
+    def __init__(self, srcpath, bindpath, recursive=False, options=None):
         MountPoint.__init__(self, mountsource=srcpath, mountpath=bindpath)
         self.srcpath = srcpath
         self.bindpath = bindpath
+        self.recursive = recursive
+        self.options = options
         self.mounted = self.ismounted()
 
     @traceLog()
     def mount(self):
         if not self.mounted:
-            util.mkdirIfAbsent(self.bindpath)
-            cmd = ['/bin/mount', '-n', '--bind', self.srcpath, self.bindpath]
+            if os.path.isdir(self.srcpath):
+                util.mkdirIfAbsent(self.bindpath)
+            elif not os.path.exists(self.bindpath):
+                util.touch(self.bindpath)
+            cmd = ['/bin/mount', '-n']
+            if self.recursive:
+                cmd.append('--rbind')
+            else:
+                cmd.append('--bind')
```

Comment:
Btw., why `--bind` doesn't work and `--rbind` does? (considering that we don't actually need recursive mounts inside chroot for the package builds).

Reply:
See above about why we need to --rbind for the /proc and /sys mounts.

Reply:
Honestly, the question isn't answered here, moving here https://bugzilla.redhat.com/show_bug.cgi?id=1745048